primitives/traits/src/lib.rs

```diff
@@ -172,7 +192,7 @@ pub trait RemoveInvulnerables<AccountId> {
 pub trait RemoveParaIdsWithNoCredits {
     /// Remove para ids with not enough credits. The resulting order will affect priority: the first para id in the list
     /// will be the first one to get collators.
-    fn remove_para_ids_with_no_credits(para_ids: &mut Vec<ParaId>);
+    fn remove_para_ids_with_no_credits(para_ids: &mut Vec<ParaId>, old_assigned: &Vec<ParaId>);
```

Comment:
This should make `.contains()` a bit faster when we have many para ids

```suggestion
    fn remove_para_ids_with_no_credits(para_ids: &mut Vec<ParaId>, old_assigned: &BTreeSet<ParaId>);
```